static/app/views/issueDetails/streamline/foldSection.tsx

```diff
@@ -0,0 +1,170 @@
+import styled from '@emotion/styled';
+
+import ErrorBoundary from 'sentry/components/errorBoundary';
+import InteractionStateLayer from 'sentry/components/interactionStateLayer';
+import {IconChevron} from 'sentry/icons';
+import {space} from 'sentry/styles/space';
+import {useLocalStorageState} from 'sentry/utils/useLocalStorageState';
+
+const LOCAL_STORAGE_PREFIX = 'fold-section-collapse-';
+export const enum FoldSectionKey {
+  // Suspect Commits & Traces
+  USER_FEEDBACK = 'issue-details-user-feedback', // In development
+  LLM_MONITORING = 'issue-details-llm-monitoring',
+
+  UPTIME = 'issue-details-uptime', // Only Uptime issues
+  CRON = 'issue-details-cron-timeline', // Only Cron issues
+
+  HIGHLIGHTS = 'issue-details-highlights',
+  RESOURCES = 'issue-details-resources', // Position controlled by flag
+
+  EVIDENCE = 'issue-details-evidence',
+  MESSAGE = 'issue-details-message',
+  STACK_TRACE = 'issue-details-stack-trace',
+
+  THREADS = 'issue-details-threads',
+  THREAD_STATE = 'issue-details-thread-state',
+  THREAD_TAGS = 'issue-details-thread-tags',
+
+  // QuickTraceQuery -> todo
+
+  SPAN_EVIDENCE = 'issue-details-span-evidence',
+  HYDRATION_DIFF = 'issue-details-hydration-diff',
+  REPLAY = 'issue-details-replay',
+
+  HPKP = 'issue-details-hpkp',
+  CSP = 'issue-details-csp',
+  EXPECTCT = 'issue-details-expectct',
+  TEMPLATE = 'issue-details-template',
+
+  BREADCRUMBS = 'issue-details-breadcrumbs',
+  DEBUGMETA = 'issue-details-debugmeta',
+  REQUEST = 'issue-details-request',
+
+  TAGS = 'issue-details-tags',
+  SCREENSHOT = 'issue-details-screenshot',
+
+  CONTEXTS = 'issue-details-contexts',
+  EXTRA = 'issue-details-extra',
+  PACKAGE = 'issue-details-package',
+  DEVICE = 'issue-details-device',
+  VIEW_HIERARCHY = 'issue-details-view-hierarchy',
+  ATTACHMENTS = 'issue-details-attachments',
+  SDK = 'issue-details-sdk',
+  GROUPING_INFO = 'issue-details-grouping-info',
+  RRWEB = 'issue-details-rrweb',
+}
+
+interface FoldSectionProps {
+  children: React.ReactNode;
+  /**
+   * Unique key to persist user preferences for initalizing the section to open/closed
+   */
+  sectionKey: FoldSectionKey;
+  /**
+   * Title of the section, always visible
+   */
+  title: React.ReactNode;
+  /**
+   * Actions associated with the section, only visible when open
+   */
+  actions?: React.ReactNode;
+  /**
+   * Should this section be initially open, gets overridden by user preferences
+   */
+  initialCollapse?: boolean;
+  /**
+   * Disable the ability for the user to collapse the section
+   */
+  preventCollapse?: boolean;
+}
+
+export function FoldSection({
+  children,
+  title,
+  actions,
+  sectionKey,
+  initialCollapse = false,
+  preventCollapse = false,
+  ...props
+}: FoldSectionProps) {
+  const [isCollapsed, setIsCollapsed] = useLocalStorageState(
+    `${LOCAL_STORAGE_PREFIX}${sectionKey}`,
+    initialCollapse
+  );
+  const [isLayerEnabled, setIsLayerEnabled] = useState(true);
```

Comment:
Why is `isLayerEnabled` necessary?

Reply:
I wasn't sure of another way to accomplish this, but it's used to disable the interaction state layer when hovering over the 'button area', to keep things legible.

https://github.com/user-attachments/assets/7f61aabc-94d8-4806-836a-81dcb5296d15

Reply:
Is this still the case, i wasn't able to repro

https://github.com/user-attachments/assets/4aed6e83-1e77-4773-a3d6-4a29e265b9e4

Reply:
Ahh I see! If you can't find another way around that I'd add a comment.

One other thing you could try would be styling the header with something like:

```css
&:has(button:hover) {
  ${InteractionStateLayer} {
    display: none;
  }
}
```

Not sure if that actually works or not (or is any better)